src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+
+namespace Bit.Client.Web.BlazorUI
+{
+    public partial class BitNav
+    {
+        [Parameter] public string SelectedKey { get; set; }
+        [Parameter] public string AriaLabel { get; set; }
+        [Parameter] public bool IsOnTop { get; set; }
+        [Parameter] public List<NavLink> NavLinks { get; set; }
+        [Parameter] public EventCallback<NavLink> OnClick { get; set; }
+        [Parameter] public Func<NavLink, RenderFragment> OnRenderHeader { get; set; }
+        protected override string RootElementClass => "bit-nav";
+
+        protected virtual async Task Toggle(NavLink navLink)
+        {
+            if (IsEnabled is false) return;
+
+            if (navLink.Disabled) return;
+
+            if (navLink.Links?.Any() ?? false)
+            {
+                navLink.IsExpanded = !navLink.IsExpanded;
+                StateHasChanged();
+            }
+
+            await OnClick.InvokeAsync(navLink);
+        }
+
+        protected override void RegisterComponentClasses()
+        {
+            ClassBuilder.Register(() => IsOnTop ? $"{RootElementClass}-top-{VisualClassRegistrar()}" : $"{RootElementClass}-no-top-{VisualClassRegistrar()}");
+        }
+
+        private RenderFragment GenerateHeader(NavLink navLink)
+        {
+            return OnRenderHeader.Invoke(navLink);
+        }
+
+        private static string MapNavLinkTargetTypeToString(NavLinkTargetType navLinkTargetType)
+        {
+            return navLinkTargetType switch
+            {
+                NavLinkTargetType.Blank => "_blank",
+                NavLinkTargetType.Parent => "_parent",
+                NavLinkTargetType.Self => "_self",
+                NavLinkTargetType.Top => "_top",
+                _ => throw new System.Exception($"NavLinkTargetType not supported: {navLinkTargetType}")
+            };
+        }
+
+        private string GetLinkClass(NavLink navLink, string selectedKey)
```

Comment:
Is it required to pass the "selectedKey" as an argument to this method?
isn't it available here as the "SelectedKey" parameter is in this class instance?